lib/src/file_util.rs

```diff
@@ -134,6 +146,39 @@ pub fn persist_content_addressed_temp_file<P: AsRef<Path>>(
     }
 }
 
+/// Symlinks are always available on UNIX, but on Windows they require admin or
+/// Developer Mode
+#[cfg(windows)]
+pub fn try_symlink<P: AsRef<Path>, Q: AsRef<Path>>(
+    original: P,
+    link: Q,
+) -> Result<(), SymlinkError> {
+    const ERROR_PRIVILEGE_NOT_HELD: i32 = 1314;
+
+    if Path::is_dir(original.as_ref()) {
+        symlink_dir(original, link)
+    } else {
+        symlink_file(original, link)
+    }
+    .map_err(|err| {
+        if let Some(ERROR_PRIVILEGE_NOT_HELD) = err.raw_os_error() {
```

Comment:
This `if let` statement doesn't seem to bind any value, so can it be written as `if err.raw_os_error() == Some(ERROR_PRIVILEGE_NOT_HELD)`?

Reply:
Fixed. This was just me running up against the limits of my Rust knowledge - I didn't know about `Option<T>`s' `impl Eq where T : Eq`.